### PR TITLE
Refactor the finding object and base class

### PR DESCRIPTION
**Pull request by rroupski**
This PR is based on 09/07/2023 meeting discussions

Roughly based on PR #763 **The new `finding_info` object**, and it is meant to replace the PR #763.
Related issue: #762 **New `finding_info` attribute as a successor to the current finding attribute.**

The original `security_finding` class has been restored (to keep backwards compatibility).
The existing `finding` object has been deprecated.
New `finding_info` object to replace the deprecated finding object.
Rename the kill chain object.

### CODE REVIEW

**Comment by pagbabian-splunk** [comment]
Aside from my two comments, it looks good to me.

**Review by irakledibm — APPROVED** [review]
Based on our group discussion, LGTM

**Review by maxhotta — APPROVED** [review]

File: events/findings/finding.json [hunk 1]
```diff
@@ -1,20 +1,12 @@
 {
   "caption": "Finding",
   "category": "findings",
-  "description": "The Finding event is a generic event that defines a set of attributes available in the Findings category. These attributes will be inherited by all the classes defined in the Findings category.",
+  "description": "The Finding event is a generic event that defines a set of attributes available in the Findings category.",
```
Comment (floydtree) [inline events/findings/finding.json:4]: We should add this line in the description - "These attributes will be inherited by all the classes defined in the Findings category." 
Many have asked what's the use-case of this "hidden" class and having a self-evident description helps alleviate those questions.

File: events/findings/vulnerability_finding.json [hunk 2]
```diff
@@ -1,17 +1,12 @@
 {
   "caption": "Vulnerability Finding",
-  "category": "findings",
-  "description": "Vulnerability Finding events describe findings generated by vulnerability scanning tools.",
+  "description": "The Vulnerability Finding event is a notification about weakness in an information system, system security procedures, internal controls, or implementation that could be exploited or triggered by a threat source.",
   "extends": "finding",
   "name": "vulnerability_finding",
   "uid": 2,
   "attributes": {
-    "analytic": {
```
Comment (floydtree) [inline events/findings/vulnerability_finding.json:9]: Why remove analytics from this object? Let's not make arbitrary changes please!

Considering each vulnerability findings class, has an array of vulnerabilities, each vulnerability can have a distinct analytic method used to assess the vulnerability in question. That's why we have it here.

**Review by floydtree — CHANGES_REQUESTED** [review]

**Pull request merged**